pkg/cable/libreswan/libreswan.go

```diff
@@ -320,11 +320,14 @@ func (i *libreswan) DisconnectFromEndpoint(endpoint types.SubmarinerEndpoint) er
 func removeConnectionForEndpoint(connections []subv1.Connection, endpoint types.SubmarinerEndpoint) []subv1.Connection {
 	for j := range connections {
 		if connections[j].Endpoint.CableName == endpoint.Spec.CableName {
-			if j == 0 && j == len(connections)-1 {
+			if len(connections) == 1 {
+				// Removing the only connection
 				return []subv1.Connection{}
 			} else if j == 0 {
+				// Removing the first connection
 				return connections[j+1:]
 			} else if j == len(connections)-1 {
+				// Removing the last connection
 				return connections[:j]
 			}
 			return append(connections[:j], connections[j+1:]...)
```

Comment:
I think simpler way to do this:

```suggestion
			copy(connections[j:], connections[j+1:])
			return connections[:len(connections)-1]
```

Reply:
Thanks, this is much better!